pkg/artifactory/resource_xray_policy.go

```diff
@@ -493,6 +498,11 @@ func resourceXrayPolicyDelete(d *schema.ResourceData, m interface{}) error {
 	c := m.(*ArtClient).Xray
 
 	resp, err := c.V1.Policies.DeletePolicy(context.Background(), d.Id())
+
+	if resp == nil {
```

Comment:
Check `err` first. If you check error first you probably never need to check if `resp` is `nil`.